src/thunderbots/software/network_input/backend.cpp

```diff
@@ -1,149 +1,224 @@
 #include "network_input/backend.h"
 
-#include "network_input/util/ros_messages.h"
 #include "proto/messages_robocup_ssl_detection.pb.h"
 #include "proto/messages_robocup_ssl_geometry.pb.h"
 #include "shared/constants.h"
 #include "util/constants.h"
 
-Backend::Backend() : ball_filter(), friendly_team_filter(), enemy_team_filter() {}
+// We can initialize the field_state with all zeroes here because this state will never
+// be accessed by an external observer to this class. the getFieldData must be called to
+// get any field data which will update the state with the given protobuf data
+Backend::Backend()
+    : field_state(0, 0, 0, 0, 0, 0, 0),
+      ball_state(Point(), Vector(), Timestamp::fromSeconds(0)),
+      friendly_team_state(Duration::fromMilliseconds(
+          Util::Constants::ROBOT_DEBOUNCE_DURATION_MILLISECONDS)),
+      enemy_team_state(Duration::fromMilliseconds(
+          Util::Constants::ROBOT_DEBOUNCE_DURATION_MILLISECONDS)),
+      ball_filter(),
+      friendly_team_filter(),
+      enemy_team_filter()
+{
+}
 
-std::optional<thunderbots_msgs::Field> Backend::getFieldMsg(
-    const SSL_WrapperPacket &packet)
+Field Backend::getFieldData(const SSL_WrapperPacket &packet)
 {
-    if (packet.has_geometry())
+    if (packet.has_geometry() && packet.geometry().has_field())
+    {
+        SSL_GeometryFieldSize field_data = packet.geometry().field();
+        Field field                      = createFieldFromPacketGeometry(field_data);
+
+        field_state = field;
+    }
+
+    return field_state;
+}
+
+Field Backend::createFieldFromPacketGeometry(
+    const SSL_GeometryFieldSize &packet_geometry) const
+{
+    // We can't guarantee the order that any geometry elements are passed to us in, so
+    // We map the name of each line/arc to the actual object so we can refer to them
+    // consistantly
+    std::map<std::string, SSL_FieldCicularArc> ssl_circular_arcs;
+    std::map<std::string, SSL_FieldLineSegment> ssl_field_lines;
+
+    // Circular arcs
+    //
+    // Arc names:
+    // CenterCircle
+    for (int i = 0; i < packet_geometry.field_arcs_size(); i++)
+    {
+        const SSL_FieldCicularArc &arc = packet_geometry.field_arcs(i);
+        std::string arc_name           = arc.name();
+        ssl_circular_arcs[arc_name]    = arc;
+    }
+
+    // Field Lines
+    //
+    // Line names:
+    // TopTouchLine
+    // BottomTouchLine
+    // LeftGoalLine
+    // RightGoalLine
+    // HalfwayLine
+    // CenterLine
+    // LeftPenaltyStretch
+    // RightPenaltyStretch
+    // RightGoalTopLine
+    // RightGoalBottomLine
+    // RightGoalDepthLine
+    // LeftGoalTopLine
+    // LeftGoalBottomLine
+    // LeftGoalDepthLine
+    // LeftFieldLeftPenaltyStretch
+    // LeftFieldRightPenaltyStretch
+    // RightFieldLeftPenaltyStretch
+    // RightFieldRightPenaltyStretch
```

Comment:
I agree it's not ideal, but I don't think an enum works in this case. These field lines/circles are only distinguishable by their `name` field:

```
  // Name of this field marking.
  required string name = 1;
```

And there's no explicit guarantee the shapes will be published in the same order allowing us to index them with an enum. So we determined the strings through observation/testing and have that list in the comments to it's easy to access other lines/shapes in the future.

Reply:
Hmm. Could we at least populate a struct with the lines we need and throw an exception/warn/skip the current iteration if we're missing some?

Reply:
The struct seems like an almost pointless middelman here. 

I can check for the existence of the keys and catch/throw any exceptions due to missing keys (although we'll already get an exception if the key isn't found). We wouldn't want to skip the iteration in this case, since it would be slightly messy to propagate an optional all the way back up to the main loop, and because this kind of error indicates we are not handling the protocol correctly which is not something we should tolerate.

Reply:
Good argument. I'm fine with leaving this as-is then, I imagine any exception a missing key generates should be pretty easy to trace.